cassandra/protocol.py

```diff
@@ -768,14 +780,38 @@ class PrepareMessage(_MessageType):
     opcode = 0x09
     name = 'PREPARE'
 
-    def __init__(self, query):
+    def __init__(self, query, keyspace=None):
         self.query = query
+        self.keyspace = keyspace
 
     def send_body(self, f, protocol_version):
         write_longstring(f, self.query)
+
+        flags = 0x00
+
+        if self.keyspace is not None:
+            if ProtocolVersion.uses_keyspace_flag(protocol_version):
+                flags |= _WITH_KEYSPACE_FLAG
```

Comment:
There's a small bug here, I only noticed it because I made the same mistake in java driver.  When sending a prepare message, the flag value for keyspace is `0x01` not `0x80` like it is for other message types.

Reply:
Whoooooops, thanks Andy